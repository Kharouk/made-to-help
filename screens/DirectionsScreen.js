import React from 'react';
import { AppRegistry, Text, TouchableOpacity, TextInput, View, StyleSheet } from 'react-native';
import { Button } from 'react-native';
import { createStackNavigator } from 'react-navigation';
import MapScreen from '../screens/MapScreen';

const App = createStackNavigator({
  Home: MapScreen,
});

export default class Directions extends React.Component {
  static navigationOptions = {
    title: 'Directions',
  };

  constructor(props) {
    super(props);
    this.state = { text: 'Destination' };
  }

  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style = { styles.container }>
            <TextInput style ={ styles.input }
            underLineColorAndroid = "transparent"
            placeholder = "Enter Destination"
            placeholderTextColor = "grey"
            onChangeText={(text) => this.setState({text})}
            value={this.state.text}
            />
          <TouchableOpacity
            style = {styles.submitButton}
            onPress={() => navigate('Home')}
          >
            <Text style = {styles.submitButtonText}> Search </Text>
          </TouchableOpacity>
      </View>
    );
  }
}
    const styles = StyleSheet.create({
     container: {
        paddingTop: 23
     },
     input: {
        width: window.width,
        padding: 4,
        margin: 10,
        height: 40,
        borderWidth: 4,
        borderColor: '#7a42f4',
        textAlign: 'center',
        alignItems:'center',
        justifyContent:'center',
        backgroundColor:'#fff',
        borderColor:'#888',
        borderRadius: 10
     },
     submitButton: {
        backgroundColor: 'rgb(0, 122, 255)',
        padding: 4,
        margin: 10,
        height: 40,
        alignItems:'center',
        justifyContent:'center',
        borderRadius: 10
     },
     submitButtonText:{
        color: 'white'
     }
  })
