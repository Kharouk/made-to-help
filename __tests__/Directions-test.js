import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import DirectionsScreen from '../screens/DirectionsScreen';

describe('Directions snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders directions form', async () => {
    const tree = renderer.create(<DirectionsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
