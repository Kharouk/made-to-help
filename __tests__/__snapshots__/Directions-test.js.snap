// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directions snapshot renders directions form 1`] = `
<View
  style={
    Object {
      "paddingTop": 23,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Start"
    placeholderTextColor="grey"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#888",
        "borderRadius": 10,
        "borderWidth": 4,
        "height": 40,
        "justifyContent": "center",
        "margin": 10,
        "padding": 4,
        "textAlign": "center",
      }
    }
    underLineColorAndroid="transparent"
    underlineColorAndroid="transparent"
    value="Destination"
  />
  <TextInput
    allowFontScaling={true}
    placeholder="End"
    placeholderTextColor="grey"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#888",
        "borderRadius": 10,
        "borderWidth": 4,
        "height": 40,
        "justifyContent": "center",
        "margin": 10,
        "padding": 4,
        "textAlign": "center",
      }
    }
    underLineColorAndroid="transparent"
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(0, 122, 255)",
        "borderRadius": 10,
        "height": 40,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
       Search 
    </Text>
  </View>
</View>
`;
